reconcile/ocm_clusters.py

```diff
@@ -24,6 +24,42 @@ def fetch_desired_state(clusters):
     return desired_state
 
 
+def get_cluster_update_spec(cluster_name, current_spec, desired_spec):
+    """ Get a cluster spec to update. Returns an error if diff is invalid """
+    allowed_spec_update_fields = [
+        'instance_type',
+        'storage',
+        'load_balancers',
+        'private',
+        'channel',
+        'autoscale',
+        'nodes'
+    ]
+    error = False
+    if current_spec['network'] != desired_spec['network']:
+        error = True
+        logging.error(f'[{cluster_name}] invalid update: network')
+    current_spec_spec = current_spec['spec']
+    desired_spec_spec = desired_spec['spec']
+    updated = {k: desired_spec_spec[k] for k in desired_spec_spec
+               if current_spec_spec[k] != desired_spec_spec[k]}
+
+    # we only need deleted to check if a field removal is valid
+    # we really want to check updated + deleted, and since
+    # we have no further use for deleted -
+    deleted = {k: current_spec_spec[k] for k in current_spec_spec
```

Comment:
You're not using the values of this dictionary at all, why not have this a set of keys?

```python
deleted = {k for k in current_spec_spec...}
```

Reply:
we are using them further down the road when actually updating the cluster